### ISSUE TITLE
Travis Ci caching (via ccache)

### ISSUE THREAD

**Issue by nmaier** [aria2/aria2#710]
It might be worth considering to enable `cache: ccache` and whatever else is required to speed up travis builds

**Comment by tatsuhiro-t** [comment]
Have you checked that it speeds up build?

**Comment by nmaier** [comment]
It does for me when I do `make; make clean; make`, enormously. Since most commits will not invalidate a lot of units and travis says it shares the cache between builds as far as I understand, it should speed up builds too; only compile what was changed instead of everything
https://docs.travis-ci.com/user/caching/#Cache-content-can-be-accessed-by-pull-requests

**Comment by nmaier** [comment]
I just tried it...
Did an initial cached-enabled build, build time was about 1min slower since the cache was empty, this is expected and more or less the worst case.
Then I pushed again, changing a single ".cc" file, which is kinda the best case: https://travis-ci.org/aria2/aria2/builds/145922491
The build for `gcc` went from about 23min to about 4min and for `clang` from about 14min to about 4min. This is better than expected.
For somewhat more involved commits (e.g. changing a common header) the savings will less but still sizable most of the time
